src/m365/entra/commands/pim/pim-role-assignment-add.ts

```diff
@@ -0,0 +1,319 @@
+      },
+      {
+        option: '-i, --roleDefinitionId [roleDefinitionId]'
+      },
+      {
+        option: "--userId [userId]"
+      },
+      {
+        option: "--userName [userName]"
+      },
+      {
+        option: "--groupId [groupId]"
+      },
+      {
+        option: "--groupName [groupName]"
+      },
+      {
+        option: "--administrativeUnitId [administrativeUnitId]"
+      },
+      {
+        option: "--applicationId [applicationId]"
+      },
+      {
+        option: "-j, --justification [justification]"
+      },
+      {
+        option: "-s, --startDateTime [startDateTime]"
+      },
+      {
+        option: "-e, --endDateTime [endDateTime]"
+      },
+      {
+        option: "-d, --duration [duration]"
+      },
+      {
+        option: "--ticketNumber [ticketNumber]"
+      },
+      {
+        option: "--ticketSystem [ticketSystem]"
+      },
+      {
+        option: "--no-expiration"
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.roleDefinitionId && !validation.isValidGuid(args.options.roleDefinitionId)) {
+          return `${args.options.roleDefinitionId} is not a valid GUID`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.startDateTime && !validation.isValidISODateTime(args.options.startDateTime)) {
+          return `${args.options.startDateTime} is not a valid ISO 8601 date time string`;
+        }
+
+        if (args.options.endDateTime && !validation.isValidISODateTime(args.options.endDateTime)) {
+          return `${args.options.endDateTime} is not a valid ISO 8601 date time string`;
+        }
+
+        if (args.options.duration && !validation.isValidISODuration(args.options.duration)) {
+          return `${args.options.duration} is not a valid ISO 8601 duration`;
+        }
+
+        if (args.options.administrativeUnitId && !validation.isValidGuid(args.options.administrativeUnitId)) {
+          return `${args.options.administrativeUnitId} is not a valid GUID`;
+        }
+
+        if (args.options.applicationId && !validation.isValidGuid(args.options.applicationId)) {
+          return `${args.options.applicationId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['roleDefinitionName', 'roleDefinitionId'] });
+    this.optionSets.push({
+      options: ['noExpiration', 'endDateTime', 'duration'],
+      runsWhen: (args) => {
+        return !args.options.noExpiration && args.options.endDateTime !== undefined || args.options.duration !== undefined;
```

Comment:
Could you take a second look at this condition? It seems incorrect to me.

Reply:
This condition should be also changed.